stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/content/MediaAttachmentPreviewContent.kt

```diff
@@ -0,0 +1,116 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.unit.dp
+import com.getstream.sdk.chat.utils.extensions.imagePreviewUrl
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.compose.ui.components.CancelIcon
+import io.getstream.chat.android.compose.ui.components.composer.MessageInput
+import io.getstream.chat.android.compose.ui.util.rememberStreamImagePainter
+import io.getstream.chat.android.uiutils.constant.AttachmentType
+
+/**
+ * UI for currently selected image and video attachments, within the [MessageInput].
+ *
+ * @param attachments Selected attachments.
+ * @param onAttachmentRemoved Handler when the user removes an attachment from the list.
+ * @param modifier Modifier for styling.
+ */
+@Composable
+public fun MediaAttachmentPreviewContent(
+    attachments: List<Attachment>,
+    onAttachmentRemoved: (Attachment) -> Unit,
+    playButton: @Composable () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    LazyRow(
+        modifier = modifier.clip(RoundedCornerShape(topStart = 16.dp, topEnd = 16.dp)),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.spacedBy(4.dp, Alignment.Start)
+    ) {
+        items(attachments) { image ->
+            MediaAttachmentPreviewItem(
+                mediaAttachment = image,
+                onAttachmentRemoved = onAttachmentRemoved,
+                playButton = playButton
+            )
+        }
+    }
+}
+
+/**
+ * A preview of an individual selected image or video attachment.
+ *
+ * @param mediaAttachment The selected attachment.
+ * @param onAttachmentRemoved Handler when the user removes an attachment from the list.
+ * @param playButton Represents the play button that is overlaid above video attachment
+ * previews.
+ */
+@Composable
+private fun MediaAttachmentPreviewItem(
+    mediaAttachment: Attachment,
+    onAttachmentRemoved: (Attachment) -> Unit,
+    playButton: @Composable () -> Unit,
+) {
+    val painter = rememberStreamImagePainter(data = mediaAttachment.upload ?: mediaAttachment.imagePreviewUrl)
+
+    Box(
+        modifier = Modifier
+            .size(MediaAttachmentPreviewItemSize.dp)
+            .clip(RoundedCornerShape(16.dp)),
+        contentAlignment = Alignment.Center
+    ) {
+        Image(
+            modifier = Modifier.fillMaxSize(),
+            painter = painter,
+            contentDescription = null,
+            contentScale = ContentScale.Crop
+        )
+
+        if (mediaAttachment.type == AttachmentType.VIDEO) {
```

Comment:
does this button trigger any actions if we preview it though?

Reply:
No, it's only used as a trigger in the gallery, elsewhere it's just a common icon.